scripts/commands/tag.js

```diff
@@ -0,0 +1,82 @@
+import fs from 'fs';
+import chalk from 'chalk';
+
+import {
+	__dirname,
+	error,
+	info,
+	getCurrentVersion,
+	getCurrentBranchName,
+	gitFactory,
+	promptContinue,
+	switchToBranchWithMessage,
+	tagExists,
+	verifyBuild,
+	warning,
+} from '../utils.js';
+
+async function tagRelease() {
+	// Ensure we're always running in the project root.
+	process.chdir( `${ __dirname }/..` );
+
+	const verificationErrors = await verifyBuild();
+	if ( verificationErrors.length > 0 ) {
+		error( 'Build verification failed:' );
+		verificationErrors.map( ( err ) => error( `\t${ err }` ) );
+		return false;
+	}
+
+	if ( fs.existsSync( 'node_modules' ) ) {
+		error(
+			'A node_modules folder exists. Please remove it and try again.'
+		);
+		return false;
+	}
+
+	let res = null;
+	const git = gitFactory();
+	const currentBranchName = await getCurrentBranchName();
+	const version = getCurrentVersion();
+	const versionStr = `${ chalk.blue( version ) }`;
+
+	res = await tagExists( version );
+	if ( res ) {
+		error( `The tag ${ version } already exists. Deploy failed.` );
```

Comment:
After getting this error my working copy was on the `master` branch instead of the `1101-update/tag-release-build` branch. I don't know if this process is meant to be run on a different branch than `master` but it felt messy to be on a different branch after an error.
<img width="1569" alt="image" src="https://user-images.githubusercontent.com/375980/235941629-ae942b2f-8a41-4658-bee1-a87151543e1f.png">

Reply:
This is a bug! You're supposed to end up back at the original branch you started on after an error.

The issue here is that we switched to `master` as part of the build step. The way it's written now, the tagging step thinks the "original branch" _is_ `master`.

I'll update with a fix.